### ISSUE TITLE
Default values for bamUtil trimming

### ISSUE THREAD

**Issue by scarlhoff** [nf-core/eager#612]
Hi!

I would like to suggest to change the default values for `bamutils_clip_none_udg_left/right` from 1 to 0. 

As bam trimming is a feature you can turn on if you want to use it, it would be more logical to put the default to 0 to avoid unwanted bam trimming, particularly when the user wants to trim one type of library only. 
Additionally, the single-stranded mode of pileupCaller doesn't require trimming of single-stranded not UDG treated data to remove damage anymore, so "accidental" trimming of bases would needlessly reduce the number of SNPs.

Thanks!

**Comment by jfy133** [comment]
Thoughts @nf-core/eager ?

**Comment by apeltzer** [comment]
Doesn't harm to have a safe default, which 0 would be a good choice as it doesn't modify what you're doing. I agree that setting the clip parameter intrinsically should make it clear that one needs to set a value (could also reflect that in the docs, if we didn't do that already). 

tl; dr: Yes, makes sense to me :-)

**Comment by jfy133** [comment]
@TCLamnidis does this make sense to you? Do we habve experience with setting bam trimming to `0`?

**Comment by TCLamnidis** [comment]
I discussed this with @scarlhoff a bit the other day. On one hand, I feel like turning trimming on should turn it on. On the other hand, a safe default is a good idea. I think I have switched sides since our discussion, and would now favour a default of `0`. 

I can't say I have experience with it. I just tried running a bam through while trimming `0` bases on either side and ran into no errors, so a quick swap in the defaults is unlikely to destroy the pipeline.

**Comment by TCLamnidis** [comment]
Might be worthwhile to implement a small warning in cases where trimming is on but all trimming parameters are left to 0 too.

**Comment by jfy133** [comment]
@TCLamnidis can you quickly checked reads weren't actualy modified, even if there were no errors?

**Comment by TCLamnidis** [comment]
#### diff
```
$ diff -sq <(samtools view -h BOO002.A0101.TF1.1_rmdup.bam) <(samtools view -h BOO002.trimmed.bam )
Files /dev/fd/63 and /dev/fd/62 are identical
```
#### md5sum
```
$ md5sum BOO002.A0101.TF1.1_rmdup.bam BOO002.trimmed.bam 
030d029d5383054c82b23c2aebd739ae  BOO002.A0101.TF1.1_rmdup.bam
7e30527e57a64f010e74f144d8b8b740  BOO002.trimmed.bam
```
Some change occurs, but the content of the file itself seems fine. My guess is that trimming outputs a bam of different compression than I started with.

**Comment by jfy133** [comment]
Prima, will schedule this for the next release.
